src/main/java/com/google/sps/managers/DatastoreManager.java

```diff
@@ -385,25 +386,39 @@ public User retrieveUserByUsername(String username) {
   public void createOrReplaceIdentification(String email, String id) {
     Entity identificationEntity = new Entity ("Identification", email);
     identificationEntity.setProperty("id", id);
+
     datastore.put(identificationEntity);
   }
   
   /**
   * Retrieves an id of a user in datastore by the email address
   * @param email a String variable representing a user's email address
+  * @throws EntityNotFoundException an exception thrown when an entity is not found
   */
-  public String retrieveIdByEmail(String email) {
+  public String retrieveIdByEmail(String email) throws EntityNotFoundException {
     Key idEntityKey = KeyFactory.createKey("Identification", email);
     Entity idEntity;
-    try {
-      idEntity = datastore.get(idEntityKey);
-    }
-    catch(Exception e) {
-      return null;
-    }
+    idEntity = datastore.get(idEntityKey);
     
     String id = (String) idEntity.getProperty("id");
     return id;
   }
+
+  public Map<String, String> retrieveKeys()
+  {
+    Map<String, String> apiKeys = new HashMap<>();
+    
+    Query query = new Query("apiKeys");
+    PreparedQuery pq = datastore.prepare(query);
+    Entity userEntity = pq.asSingleEntity();
+
+    String clientId = (String) userEntity.getProperty("CLIENT_ID");
+    String mapApi = (String) userEntity.getProperty("MAP_key");
```

Comment:
Should probably capitalize 'key' here to keep things consistent.

Reply:
done

Reply:
done